cmd/create/oidcprovider/cmd.go

```diff
@@ -188,7 +194,7 @@ func run(cmd *cobra.Command, _ []string) {
 	}
 
 	mode := args.mode
-	if interactive.Enabled() {
+	if interactive.Enabled() && !hasMode(mode) {
```

Comment:
You can probably just do `if interactive.Enabled() && !skipInteractive` (or maybe just `if interactive.Enabled() && mode == ""`)